lua/iswap.lua

```diff
@@ -93,8 +93,12 @@ function M.iswap_node_with(direction, config)
     if outer_row ~= current_row or outer_col ~= current_col then -- new outer parent to have same start()
       break
     end
-    if util.has_siblings(outer_cursor_node) then
+    if direction == 'right' and outer_cursor_node:next_named_sibling() ~= nil then -- only select node if it has a right sibling
       last_valid_node = outer_cursor_node
+    elseif direction == 'left' and outer_cursor_node:prev_named_sibling() ~= nil then  -- or left sibling
+      last_valid_node = outer_cursor_node
+    elseif direction == nil and util.has_siblings(outer_cursor_node) then  -- if no direction, then node with any sibling is ok
+        last_valid_node = outer_cursor_node
```

Comment:
sorry i think you've missed this one

Reply:
Ok I think I got it all right in last commit.